File: Programmers/내일은 코딩테스트 with 파이썬/파트 05, 집합(셋)/폰켓몬.py
def solution(nums):
    answer = 0
    dict = {}
    # 골라야 할 폰켓몬의 개수
    n = len(nums)//2
    for i in nums:
        try:
            dict[i] += 1
        except:
            dict[i] = 1
    
    if len(dict) > n:
        answer = n
    else:
        answer = len(dict)
        
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>
using namespace std;

int solution(vector<int> nums)
{
    set<int> setNumbers;
    int answer = 0;
    
    for(int num:nums){
        setNumbers.insert(num);
    }
    
    int phoneketmon_size = nums.size()/2;
    
    if(phoneketmon_size >= setNumbers.size()){ 
        answer = setNumbers.size();
    }
    else{
        answer = phoneketmon_size;
    }
    
    return answer;
}